File: RecyclerViewJava/app/src/main/java/com/devwilly/recyclerview_java/MainActivity.java
package com.devwilly.recyclerview_java;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.default_recycler_view_btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openActivity(DefaultRecyclerViewActivity.class);
            }
        });
    }

    private void openActivity(Class<? extends AppCompatActivity> cls) {
        Intent intent = new Intent();
        intent.setClass(MainActivity.this, cls);
        startActivity(intent);
    }
}

File: RecyclerView-Kotlin/app/src/main/java/com/devwilly/recyclerviewkotlin/MainActivity.kt
package com.devwilly.recyclerviewkotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.default_recycler_view_btn).setOnClickListener { openActivity(DefaultRecyclerViewActivity::class.java) }
    }

    private fun openActivity(cls: Class<*>) {
        val intent = Intent()
        intent.setClass(this@MainActivity, cls)
        startActivity(intent)
    }
}
